File: frontend/web/monorepo/libs/mateu/src/mateu/shared/apiClients/dtos/componentmetadata/EntityHeader.ts
import ComponentMetadata from "@mateu/shared/apiClients/dtos/ComponentMetadata";
import Chip from "@mateu/shared/apiClients/dtos/componentmetadata/Chip";
import Fact from "@mateu/shared/apiClients/dtos/componentmetadata/Fact";

export default interface EntityHeader extends ComponentMetadata {
    title?: string
    badges?: Chip[]
    subtitle?: string
    facts?: Fact[]
    metricLabel?: string
    metricValue?: string
    metricCaption?: string
}

File: backend/shared/uidl/src/main/java/io/mateu/uidl/data/EntityHeader.java
package io.mateu.uidl.data;

import io.mateu.uidl.fluent.Component;
import java.util.List;
import lombok.Builder;

/**
 * A persistent context banner for the entity a flow operates on (e.g. the guest of a check-in
 * wizard): the {@code title} with its {@code badges} ({@link Chip}s), a {@code subtitle}, key
 * {@code facts} ({@link Fact} label-over-value pairs) and one highlighted metric on the right
 * ({@code metricLabel}, big accent-colored {@code metricValue}, {@code metricCaption}). Read-only,
 * no actions. Design-system neutral, dark-mode aware.
 */
@Builder
public record EntityHeader(
    String id,
    String title,
    List<Chip> badges,
    String subtitle,
    List<Fact> facts,
    String metricLabel,
    String metricValue,
    String metricCaption,
    String style,
    String cssClasses)
    implements Component {}
